Source:   http://bazaar.launchpad.net/~inkscape.dev/inkscape/trunk/revision/9267
Upstream: yes
Reason:   Build failure with gcc 4.5.0; fixes direct calls of constructors

=== modified file 'src/eraser-context.cpp'
--- src/eraser-context.cpp	2009-03-13 17:13:24 +0000
+++ src/eraser-context.cpp	2010-04-01 08:14:55 +0000
@@ -749,7 +749,7 @@
                 if ( eraserMode ) {
                     toWorkOn = sp_document_partial_items_in_box(sp_desktop_document(desktop), desktop->dkey, bounds);
                 } else {
-                    Inkscape::Rubberband::Rubberband *r = Inkscape::Rubberband::get(desktop);
+                    Inkscape::Rubberband *r = Inkscape::Rubberband::get(desktop);
                     toWorkOn = sp_document_items_at_points(sp_desktop_document(desktop), desktop->dkey, r->getPoints());
                 }
                 toWorkOn = g_slist_remove( toWorkOn, acid );

=== modified file 'src/extension/internal/filter/filter-file.cpp'
--- src/extension/internal/filter/filter-file.cpp	2008-11-21 05:24:08 +0000
+++ src/extension/internal/filter/filter-file.cpp	2010-04-01 08:14:55 +0000
@@ -161,7 +161,7 @@
 	mywriter writer;
 	sp_repr_write_stream(node, writer, 0, FALSE, g_quark_from_static_string("svg"), 0, 0);

-    Inkscape::Extension::build_from_mem(xml_str, new Filter::Filter(g_strdup(writer.c_str())));
+    Inkscape::Extension::build_from_mem(xml_str, new Filter(g_strdup(writer.c_str())));
 	g_free(xml_str);
     return;
 }

=== modified file 'src/extension/internal/filter/filter.cpp'
--- src/extension/internal/filter/filter.cpp	2010-01-20 23:10:33 +0000
+++ src/extension/internal/filter/filter.cpp	2010-04-01 08:14:55 +0000
@@ -219,7 +219,7 @@
                 "<menu-tip>%s</menu-tip>\n"
             "</effect>\n"
         "</inkscape-extension>\n", name, id, submenu, tip);
-    Inkscape::Extension::build_from_mem(xml_str, new Filter::Filter(filter));
+    Inkscape::Extension::build_from_mem(xml_str, new Filter(filter));
 	g_free(xml_str);
     return;
 }

=== modified file 'src/flood-context.cpp'
--- src/flood-context.cpp	2009-08-06 14:17:17 +0000
+++ src/flood-context.cpp	2010-04-01 08:14:55 +0000
@@ -900,7 +900,7 @@
     if (is_point_fill) {
         fill_points.push_back(Geom::Point(event->button.x, event->button.y));
     } else {
-        Inkscape::Rubberband::Rubberband *r = Inkscape::Rubberband::get(desktop);
+        Inkscape::Rubberband *r = Inkscape::Rubberband::get(desktop);
         fill_points = r->getPoints();
     }

@@ -1206,7 +1206,7 @@

     case GDK_BUTTON_RELEASE:
         if (event->button.button == 1 && !event_context->space_panning) {
-            Inkscape::Rubberband::Rubberband *r = Inkscape::Rubberband::get(desktop);
+            Inkscape::Rubberband *r = Inkscape::Rubberband::get(desktop);
             if (r->is_started()) {
                 // set "busy" cursor
                 desktop->setWaitingCursor();

=== modified file 'src/gradient-context.cpp'
--- src/gradient-context.cpp	2010-03-08 01:31:49 +0000
+++ src/gradient-context.cpp	2010-04-01 08:14:55 +0000
@@ -650,7 +650,7 @@
                 if (!event_context->within_tolerance) {
                     // we've been dragging, either do nothing (grdrag handles that),
                     // or rubberband-select if we have rubberband
-                    Inkscape::Rubberband::Rubberband *r = Inkscape::Rubberband::get(desktop);
+                    Inkscape::Rubberband *r = Inkscape::Rubberband::get(desktop);
                     if (r->is_started() && !event_context->within_tolerance) {
                         // this was a rubberband drag
                         if (r->getMode() == RUBBERBAND_MODE_RECT) {

=== modified file 'src/select-context.cpp'
--- src/select-context.cpp	2010-02-05 19:27:23 +0000
+++ src/select-context.cpp	2010-04-01 08:14:55 +0000
@@ -608,7 +608,7 @@
                     }
                     sc->item = NULL;
                 } else {
-                    Inkscape::Rubberband::Rubberband *r = Inkscape::Rubberband::get(desktop);
+                    Inkscape::Rubberband *r = Inkscape::Rubberband::get(desktop);
                     if (r->is_started() && !within_tolerance) {
                         // this was a rubberband drag
                         GSList *items = NULL;

